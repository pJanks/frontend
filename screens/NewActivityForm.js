import React, { Component } from 'react';
import { Text, TextInput, View, Picker, Button, Alert } from 'react-native';
import DatePicker from '../components/DatePicker';

import ActivityPicker from '../components/ActivityPicker';

export default class NewActivityForm extends Component {
  constructor() {
    super();
  }

  render() {
    

    return (
      <View>
        <Text>Activity:</Text>
        <ActivityPicker />
        <Text>Location:</Text>
        <TextInput       
          style={{ height: 40, backgroundColor: '#fff' }}
          // onChangeText={text => onChangeText(text)}
          // value={value} 
        />
        <Text>Date:</Text>
        <DatePicker />
        <Button
          title="Create Activity"
          onPress={() => Alert.alert('Activity button pressed')}
        />
      </View>
    )
  }
}