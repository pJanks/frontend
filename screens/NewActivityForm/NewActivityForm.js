import React, { useState } from 'react';
import { Text, TextInput, View, Button, Picker, StyleSheet } from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';


export default function NewActivityForm({ navigation }) {
  const [activity, setActivity] = useState('');
  const [location, setLocation] = useState('');
  const [date, setDate] = useState(new Date(1586051730000));
  const [show, setShow] = useState(false);
  const [mode, setMode] = useState('date');

  const getCurrentDate = (date) => {
    let year = date.getFullYear();
    let month, day;
    if ((date.getMonth() + 1) < 10) {
      month = `0${date.getMonth() + 1}`
    } else {
      month = date.getMonth() + 1
    }
    if (date.getDate() < 10) {
      day = `0${date.getDate()}`
    } else {
      day = date.getDate();
    }
    this.date = year + "-" + month + "-" + day;
    return year + "-" + month + "-" + day;
  }

  const onDateChange = (event, selectedDate) => {
    const currentDate = selectedDate || date;
    setShow(false);
    setDate(currentDate);
  };

  const showMode = currentMode => {
    setShow(true);
    setMode(currentMode);
  };

  const showDatepicker = () => {
    showMode('date');
  };

  return (
    <View style={ styles.container }>
      <Text style={ styles.label }>Activity:</Text>
      <View>
        <Picker
          selectedValue={activity}
          style={ styles.picker }
          onValueChange={(itemValue) => setActivity(itemValue)}
        >
          <Picker.Item label='Select an activity...' value={0} />
          <Picker.Item label='Hiking' value={1} />
          <Picker.Item label='Climbing' value={2} />
          <Picker.Item label='Biking' value={3} />
          <Picker.Item label='Running' value={4} />
          <Picker.Item label='Kayaking' value={5} />
          <Picker.Item label='Skiing' value={6} />
          <Picker.Item label='Snowboarding' value={7} />
        </Picker>
      </View>
      <Text style={ styles.label }>Location:</Text>
      <TextInput       
        style={ styles.location }
        value={ location }
        onChangeText={(value) => setLocation(value)}
      />
      <Text style={ styles.label }>Date:</Text>
      <View>
        <Button color='green' onPress={showDatepicker} title="Select a Date" />
      </View>
      {show && (
        <DateTimePicker
          testID="dateTimePicker"
          timeZoneOffsetInMinutes={0}
          value={date}
          mode={mode}
          is24Hour={true}
          display="default"
          onChange={onDateChange}
        />
      )}
      <View style={ styles.createActivity }>
        <Button
          title="Create Activity"
          onPress={() => {
            let formatDate = getCurrentDate(date)
            //Date not currently passing. Will be sent to an API call instead and this is just testing
            navigation.navigate('UserActivityScreen', {activity, location, formatDate})}}
        />
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  label: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  location: {
    backgroundColor: '#fff',
    height: 40,
    width: 250,
  },
  createActivity: {
    margin: 20,
    width: 250,
  },
  picker: {
    height: 40,
    width: 250,
    borderColor: 'gray',
    borderWidth: 1,
    backgroundColor: '#fff',
  }
})