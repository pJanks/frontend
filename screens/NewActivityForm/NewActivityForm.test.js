import React from 'react';
import NewActivityForm from './NewActivityForm';

import renderer from 'react-test-renderer';

describe('NewActivityForm', () => {
  test('should load with props', () => {
    const mockRouteInputs = jest.fn().mockImplementation(() => {
      return {
        id: 1,
        firstName: 'John',
        lastName: 'Smith'
      }
    })
    const mockNavigation = { navigate: jest.fn() };
    const mockRoute = {params: mockRouteInputs};
    const newActivityForm = renderer.create(<NewActivityForm navigation={mockNavigation} route={mockRoute} />)
    expect(newActivityForm).toMatchSnapshot();
  });
})
