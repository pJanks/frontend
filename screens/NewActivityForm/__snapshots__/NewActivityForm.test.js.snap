// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewActivityForm should load with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#b2e1f4",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "paddingTop": 15,
      }
    }
  >
    Activity:
  </Text>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "width": 250,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "Select an activity...",
              "textColor": undefined,
              "value": 0,
            },
            Object {
              "label": "Mountain Biking",
              "textColor": undefined,
              "value": "Mountain Biking",
            },
            Object {
              "label": "Hiking",
              "textColor": undefined,
              "value": "Hiking",
            },
            Object {
              "label": "Road Biking",
              "textColor": undefined,
              "value": "Road Biking",
            },
            Object {
              "label": "Baseball",
              "textColor": undefined,
              "value": "Baseball",
            },
            Object {
              "label": "Golf",
              "textColor": undefined,
              "value": "Golf",
            },
            Object {
              "label": "Cricket",
              "textColor": undefined,
              "value": "Cricket",
            },
            Object {
              "label": "Diving",
              "textColor": undefined,
              "value": "Diving",
            },
            Object {
              "label": "Soccer",
              "textColor": undefined,
              "value": "Soccer",
            },
            Object {
              "label": "Football",
              "textColor": undefined,
              "value": "Football",
            },
            Object {
              "label": "Frisbee",
              "textColor": undefined,
              "value": "Frisbee",
            },
            Object {
              "label": "Horseback Riding",
              "textColor": undefined,
              "value": "Horseback Riding",
            },
            Object {
              "label": "Kayaking",
              "textColor": undefined,
              "value": "Kayaking",
            },
            Object {
              "label": "Stand-up Paddle Boarding",
              "textColor": undefined,
              "value": "Stand-up Paddle Boarding",
            },
            Object {
              "label": "Rafting",
              "textColor": undefined,
              "value": "Rafting",
            },
            Object {
              "label": "Snowboarding",
              "textColor": undefined,
              "value": "Snowboarding",
            },
            Object {
              "label": "Skiing",
              "textColor": undefined,
              "value": "Skiing",
            },
            Object {
              "label": "Paragliding",
              "textColor": undefined,
              "value": "Paragliding",
            },
            Object {
              "label": "Rock Climbing",
              "textColor": undefined,
              "value": "Rock Climbing",
            },
            Object {
              "label": "Running",
              "textColor": undefined,
              "value": "Running",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "paddingTop": 15,
      }
    }
  >
    Location:
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="City, State or Landmark"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 40,
        "paddingLeft": 10,
        "width": 250,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "paddingTop": 15,
      }
    }
  >
    Date:
  </Text>
  <View
    style={
      Object {
        "marginBottom": 30,
        "width": 250,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#ed950e",
              },
            ]
          }
        >
          Select a Date
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 30,
        "width": 250,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Create Activity
        </Text>
      </View>
    </View>
  </View>
</View>
`;
